python/ppo/models.py

```diff
@@ -213,10 +214,12 @@ def __init__(self, lr, brain, h_size, epsilon, max_step, normalize, num_layers):
         self.create_reward_encoder()
 
         hidden_state, hidden_visual, hidden_policy, hidden_value = None, None, None, None
-        if brain.number_observations > 0:
-            height_size, width_size = brain.camera_resolutions[0]['height'], brain.camera_resolutions[0]['width']
-            bw = brain.camera_resolutions[0]['blackAndWhite']
-            hidden_visual = self.create_visual_encoder(height_size, width_size, bw, h_size, 2, tf.nn.tanh, num_layers)
+        encoders = []
+        for i in range(brain.number_observations):
+            height_size, width_size = brain.camera_resolutions[i]['height'], brain.camera_resolutions[i]['width']
+            bw = brain.camera_resolutions[i]['blackAndWhite']
+            encoders.append(self.create_visual_encoder(height_size, width_size, bw, h_size, 2, tf.nn.tanh, num_layers))
+        hidden_visual = [tf.concat(encoders, axis=1)]
```

Comment:
I am not sure this will work in the case of continuous state : encoders is a list of list of tensors. 
`[num_streams, num_observations, h_size] `
This `tf.concat` will not work in these conditions

Reply:
Good catch! We were focusing on discrete control so we missed it :sweat_smile: 

The continuous control can be fixed by changing the `axis` argument from 1 to 2 and not making a list. Please refer to this [commit](https://github.com/Unity-Technologies/ml-agents/pull/256/commits/6173d17a76029990e5644cf0fcddccc88fcf273b). We added a couple of cameras to the 3DBall environment and it seems to be working.